### PR TITLE
IGNITE-14163 fix an issue with range predicate serialization

### PR DESCRIPTION
**Pull request by korlov42**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/CalciteQueryProcessor.java [hunk 1]
```diff
@@ -71,6 +71,9 @@
         .executor(EXECUTOR)
         .sqlToRelConverterConfig(SqlToRelConverter.config()
             .withTrimUnusedFields(true)
+            // currently SqlToRelConverter creates not optimal plan for both optimization and execution
```
Comment (ygerzhedovich) [inline modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/CalciteQueryProcessor.java:74]: let's add TODO with ticket number

File: modules/core/src/main/java/org/apache/ignite/internal/processors/query/GridQueryProcessor.java [hunk 2]
```diff
@@ -2837,7 +2837,7 @@ private void checkxEnabled() throws IgniteException {
             throw new CacheException("Execution of local SqlFieldsQuery on client node disallowed.");
 
         if (experimentalQueryEngine != null && useExperimentalSqlEngine) {
-            if (!H2_REDIRECTION_RULES.matcher(qry.getSql()).find())
+            if (H2_REDIRECTION_RULES.matcher(qry.getSql()).find())
```
Comment (ygerzhedovich) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/query/GridQueryProcessor.java:2840]: we use such a check twice. Will be good to extract a method to check it?

File: modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/CalciteQueryProcessor.java (hunk 1 above)
Comment (korlov42) [reply modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/CalciteQueryProcessor.java:74]: done
https://issues.apache.org/jira/browse/IGNITE-14277

File: modules/core/src/main/java/org/apache/ignite/internal/processors/query/GridQueryProcessor.java (hunk 2 above)
Comment (korlov42) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/query/GridQueryProcessor.java]: done

**Review by ygerzhedovich — APPROVED** [review]

File: modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/externalize/RelJson.java [hunk 3]
```diff
@@ -218,6 +223,10 @@ private static void register(ImmutableMap.Builder<String, Enum<?>> builder, Clas
             "org.apache.calcite.adapter.jdbc.",
             "org.apache.calcite.adapter.jdbc.JdbcRules$");
 
+    RelJson(RelOptCluster cluster) {
```
Comment (tledkov-gridgain) [inline modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/externalize/RelJson.java:226]: javadoc

**Review by tledkov-gridgain — APPROVED** [review]
The patch is OK with me

File: modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/externalize/RelJson.java (hunk 3 above)
Comment (korlov42) [reply modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/externalize/RelJson.java:227]: add an empty javadoc

**Pull request closed without merging**